src/test/scala/com/microsoft/hyperspace/index/CreateIndexNestedTest.scala

```diff
@@ -206,4 +203,30 @@ class CreateIndexNestedTest extends HyperspaceSuite with SQLHelper {
       lineageValues.forall(lineageRange.contains(_))
     }
   }
+
+  test("Verify index creation with StructType column.") {
+    val indexConfig = IndexConfig("index1", Seq("nested"), Seq("clicks"))
+    val indexConfig2 = IndexConfig("index2", Seq("clicks"), Seq("nested"))
+    hyperspace.createIndex(nonPartitionedDataDF, indexConfig)
+    hyperspace.createIndex(nonPartitionedDataDF, indexConfig2)
+    assert(hyperspace.indexes.where(s"name = 'index1' ").count == 1)
+    assert(hyperspace.indexes.where(s"name = 'index2' ").count == 1)
+
+    import com.microsoft.hyperspace._
+    spark.enableHyperspace
+
+    {
+      val filter = nonPartitionedDataDF.filter(col("nested").isNotNull).select("nested", "clicks")
```

Comment:
Can you check if this filter works: "nested.id == 'id1'"

Reply:
Indexing StructType columns don't make much sense, so I think it's safer to block it until there are strong use cases. Including StructType columns should be okay and it would be nice if it's supported.

By the way, I think the check for column types should be done by index implementations because different index types might support different column types.

Reply:
Seems StructType column can be used as a join column, so I think it's better not to block it..? That might cause some regression.

Reply:
Okay, that seems a valid use case.